fuse/ipns/ipns_unix.go

```diff
@@ -309,279 +271,130 @@ func (s *Node) ReadDirAll(ctx context.Context) ([]fuse.Dirent, error) {
 	return nil, fuse.ENOENT
 }
 
-func (s *Node) Read(ctx context.Context, req *fuse.ReadRequest, resp *fuse.ReadResponse) error {
-	k, err := s.Nd.Key()
+func (fi *File) Read(ctx context.Context, req *fuse.ReadRequest, resp *fuse.ReadResponse) error {
+	_, err := fi.fi.Seek(req.Offset, os.SEEK_SET)
```

Comment:
not clear from the call that the `fi.Seek()` is only changing local state vs doing something heavier. not sure best way, but maybe one way is with a promise:

```go
ctx, cancel := context.WithCancel(ctx)
defer cancel()

errOut := make(chan error, 1)
go func() {
  // do all the things...
}()

select {
case <-ctx.Done():
  return ctx.Err()
case err := <-errOut:
  return err
}
```